### ISSUE TITLE
Fix VAE hyperparameter component importance TODO

### ISSUE THREAD

**Issue by Benjamin-Lee** [Benjamin-Lee/deep-rules#208]
@rasbt, you raised this this comment (see [here](https://github.com/Benjamin-Lee/deep-rules/blob/9522ea5b6db67fd06f3a98ffd4c7ce518e7967aa/content/08.hyperparameters.md#L12) for context):

 > I am not sure this is correct, why would a larger number of, e.g., pixels make the reconstruction loss more important? I suppose this is true if we just sum over the pixel-wise differences, but if we average, e.g., using MSE, I am not convinced this is true. Please comment.

Would you mind taking the lead on resolving this? I'm not sure I fully understand either the sentence in question (which should be clarified regardless of whether it is correct) nor the question itself.

**Comment by cgreene** [comment]
I agree that it is _possible_ to do it with averaging, but I would argue that it is rarely done in practice. Averaging would add an additional step that could slow the computation.

Also, here are a number of Keras and pytorch examples, in which this is not done and these are among the first hits.

<img width="519" alt="Screen Shot 2020-06-04 at 7 37 54 AM" src="https://user-images.githubusercontent.com/542643/83752495-a2eb5d00-a636-11ea-93ea-dd8b1a7a4e5a.png">

<img width="750" alt="Screen Shot 2020-06-04 at 7 38 04 AM" src="https://user-images.githubusercontent.com/542643/83752526-ad0d5b80-a636-11ea-8465-6b49fe601517.png">

<img width="630" alt="Screen Shot 2020-06-04 at 7 38 42 AM" src="https://user-images.githubusercontent.com/542643/83752536-b1397900-a636-11ea-847c-60cd5b210f29.png">

<img width="788" alt="Screen Shot 2020-06-04 at 7 38 55 AM" src="https://user-images.githubusercontent.com/542643/83752550-b4cd0000-a636-11ea-9025-3205085840b6.png">

While we might wish that something would be trivial and desirable, it doesn't mean that it is the case :)

**Comment by cgreene** [comment]
https://pytorch.org/docs/master/nn.functional.html#binary-cross-entropy

```
reduction (string, optional) – Specifies the reduction to apply to the output: 'none' | 'mean' | 'sum'. 'none': no reduction will be applied, 'mean': the sum of the output will be divided by the number of elements in the output, 'sum': the output will be summed. Note: size_average and reduce are in the process of being deprecated, and in the meantime, specifying either of those two args will override reduction. Default: 'mean'
```
Looks like for pytorch the default does make this mean. It's less clear for me with the keras examples. I agree that I'm not sure it merits a mention as a specific case, but "look under the hood and understand what you're doing" might make sense. I didn't know the answer to this and had to go looking, and it's not clear to me what would happen in the keras case with the example code.

**Comment by pstew** [comment]
How in the weeds will this point be vs keeping the discussion at a higher level? Will we have to refer readers to code examples to get the point across?

**Comment by Benjamin-Lee** [comment]
> How in the weeds will this point be vs keeping the discussion at a higher level? Will we have to refer readers to code examples to get the point across?

Ideally not in the weeds if at all possible. I would really prefer to avoid any code examples since, as the paper ages, they are usually out of date or in a different language than what the person wants.

**Comment by cgreene** [comment]
I think we are in agreement that we should say something general about understanding the nuances of implementation without specifically discussing the nuances of a single implementation. Did I summarize that correctly?

**Comment by Benjamin-Lee** [comment]
Yes, agreed.

**Comment by Benjamin-Lee** [comment]
Ok pushed to #213. @cgreene can you review?

**Comment by cgreene** [comment]
closed by #217
